google-cloud-spanner/lib/google/cloud/spanner/pool.rb

```diff
@@ -88,29 +154,43 @@ def reset
 
         def close
           @thread.kill if @thread
-          @pool.each(&:delete_session)
-          @pool = []
-          @queue = []
+          @all_sessions.each(&:delete_session)
```

Comment:
what happens if delete_session raises an error for some session? Will it still be called for all the other sessions? If not then, you should catch that error and make sure that other sessions are still deleted.